### PR TITLE
Moved global state into ide definition

### LINKED ISSUE #1066: User global state makes test fail
Tests like `takePastStartOfToken` that tests the `deprecatedPositionInference` warning relies on user global state. If the user has set 
`hideInferenceWarning` those tests will fail. The solution is to mock global state during testing.

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
Fixes #1066

## Checklist

- [x] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [ ] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet_html)

### CODE REVIEW

**Review by auscompgeek — COMMENTED** [review]
The set of state keys we need is static, right? Could the interface just be this:
```ts
interface State {
  fontRatios: {
    widthRatio: number;
    heightRatio: number;
    fontFamily: string;
  } | undefined;

  hideInferenceWarning: boolean;
}
```

**Comment by pokey** [comment]
yeah @auscompgeek I was thinking something along these lines as well, though I'd be tempted to have the knowledge of which keys need to be globally synced incorporated in some way as well, rather than hardcoded into VSCode impl

**Comment by pokey** [comment]
@auscompgeek the only issue with an interface as simple as that is that VSCode makes the setter async, which I don't think your version allows

**Comment by auscompgeek** [comment]
Yeah, if awaiting on the setter is required then we'd need separate getter/setter methods. It doesn't look like we're awaiting any of those right now though, so we could probably get away with the simple interface for now.

**Comment by pokey** [comment]
True, tho probably useful to be able to wait until set has happened. Also, if we want to let the ide handle the global state, we might need want it to be async as it will be going across the wire from cursorless lsp

**Comment by pokey** [comment]
But even if we don't switch to that simplified interface we could still make the key strongly typed

**Comment by AndreasArvidsson** [comment]
Making the key strongly typed is probably my preference.
Unsure we want to add a field for `widthRatio` and the up their hat parameters. They feel kinda vscode specific. Maybe they should just use the old implementation and not the ide abstraction?

**Comment by AndreasArvidsson** [comment]
Could we just merged this one and then do a refactor of the interface in a follow up pr? Because at the moment I can't run tests locally.

**Comment by pokey** [comment]
- [ ] as discussed, let's move default value to definition of state key rather than passing it in

Should prob do that for our settings code as well but out of scope

**Review by auscompgeek — COMMENTED** [review]
Nice. The fake and VSCode impl just need the same dependent type treatment.

File: src/core/commandVersionUpgrades/canonicalizeAndValidateCommand.ts [hunk 1]
```diff
@@ -128,11 +126,9 @@ export async function checkForOldInference(
...
 
   if (hasOldInference) {
-    const hideInferenceWarning =
-      graph.extensionContext.globalState.get<boolean>(
-        globalStateKeys.hideInferenceWarning,
-        false
-      );
+    const hideInferenceWarning = graph.ide.globalState.get(
+      "hideInferenceWarning"
+    );
```
Comment (pokey) [inline src/core/commandVersionUpgrades/canonicalizeAndValidateCommand.ts:131]: I improved the type inference a bit, so you don't need to put the `<boolean>` anymore.  I think it's now pretty slick

File: src/core/commandVersionUpgrades/canonicalizeAndValidateCommand.ts [hunk 2]
```diff
@@ -142,10 +138,7 @@ export async function checkForOldInference(
       );
 
       if (pressed) {
-        graph.extensionContext.globalState.update(
-          globalStateKeys.hideInferenceWarning,
-          true
-        );
+        graph.ide.globalState.set("hideInferenceWarning", true);
```
Comment (pokey) [inline src/core/commandVersionUpgrades/canonicalizeAndValidateCommand.ts:141]: Note that the value is now strongly typed, so you have to put a boolean here for this key

**Review by pokey — APPROVED** [review]
Made some tweaks to strengthen the type inference.  Feel free to merge if it looks good to you

File: src/core/commandVersionUpgrades/canonicalizeAndValidateCommand.ts (hunk 1 above)
Comment (AndreasArvidsson) [reply src/core/commandVersionUpgrades/canonicalizeAndValidateCommand.ts:131]: Oh that is neat!

**Pull request merged**